topbar.js

```diff
@@ -586,6 +586,11 @@ function fixTopBar() {
     let focusIsScratch = focus && Scratch.isScratchWindow(focus)
     let fullscreen = selected && selected.fullscreen && !(focusIsScratch);
     let hideTopBar = !spaces.monitors.get(panelMonitor).showTopBar
+    // hide topbar for fullscreen scratch windows
+    if (focusIsScratch && focus.fullscreen) {
```

Comment:
One particular thing I would look for: `focusIsScratch` currently includes a check to make sure that `focus` is not null, so it's safe to check `focus.fullscreen` here. But there's no guarantee `focusIsScratch` will keep that check, so it would be easy to end up with a crash here from trying to access a property of null/undefined.

However this ends up, I'd either try to get it to match the `if (!focus) return` pattern so it's safe to use in the rest of the function, or pull things out so there's an explicit `focus && focus.fullscreen` somewhere or other.

Reply:
yes, good call tbh - it was/is quite messy and a refactor is probably in order!

I'll relook at the implementation of this and go for a cleaner approach/refactor.

Reply:
I think I'd start with something like this (if I'm guessing correctly about what `selected` vs `focus` means) and go from there. It doesn't fully address the null check issue, but it moved them closer together at least, and I think would simplify the logic below.
````diff
-let fullscreen = selected && selected.fullscreen && !(focusIsScratch);
+let fullscreen = focusIsScratch ? focus.fullscreen : selected && selected.fullscreen;
````

Reply:
Thanks @smichel17, so, I've simplified the current fixTopBar() function as a first step - and added commends detailing (my) current understading/interpretation of what is happening in the code.

Let me know if that aligns with what you're seeing/thinking (especially the diff between `selected` and `focused` window).

Using your suggested logic for `fullscreen` boolean (looks good to me!).

I've been testing this simplification (no issues so far for both tiled/scratch fullscreen windows) and will continue testing this.

Reply:
Thanks @smichel17 - please checkout the latest refactor.  Requesting a re-review.  Cheers!